Safety/Inc/mpu6050_driver.hpp

```diff
@@ -0,0 +1,87 @@
+/* USER CODE BEGIN Header */
+/**
+  ******************************************************************************
+  * @file           : main.h
+  * @brief          : Header for main.c file.
+  *                   This file contains the common defines of the application.
+  ******************************************************************************
+  * @attention
+  *
+  * <h2><center>&copy; Copyright (c) 2021 STMicroelectronics.
+  * All rights reserved.</center></h2>
+  *
+  * This software component is licensed by ST under BSD 3-Clause license,
+  * the "License"; You may not use this file except in compliance with the
+  * License. You may obtain a copy of the License at:
+  *                        opensource.org/licenses/BSD-3-Clause
+  *
+  ******************************************************************************
+  */
+/* USER CODE END Header */
+
+/* Define to prevent recursive inclusion -------------------------------------*/
+#ifndef __MAIN_H
+#define __MAIN_H
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/* Includes ------------------------------------------------------------------*/
+
+#include <stdint.h>
+
+/* Private includes ----------------------------------------------------------*/
+/* USER CODE BEGIN Includes */
+
+/* USER CODE END Includes */
+
+/* Exported types ------------------------------------------------------------*/
+/* USER CODE BEGIN ET */
+
+/* USER CODE END ET */
+
+/* Exported constants --------------------------------------------------------*/
+/* USER CODE BEGIN EC */
+
+/* USER CODE END EC */
+
+/* Exported macro ------------------------------------------------------------*/
+/* USER CODE BEGIN EM */
+
+/* USER CODE END EM */
+
+/* Exported functions prototypes ---------------------------------------------*/
+void Error_Handler(void);
+
+/* USER CODE BEGIN EFP */
+
+void imuInit ();
+struct GyroData {
+	double gyro_x;
+	double gyro_y;
+	double gyro_z;
+};
+struct GyroData readGyro ();
```

Comment:
At WARG our sensor drivers all use the singleton design pattern, so it would be great if this driver could be implemented in such a way as well. As a quick rundown, singleton is basically an OOP based pattern where you can "create" a sensor object, but you can only create one of them. Since there is only one IMU in this system, the singleton design is optimal. 

https://github.com/UWARG/ZeroPilot-SW/blob/devel/Autopilot/Inc/altimeter.hpp
https://github.com/UWARG/ZeroPilot-SW/blob/devel/Autopilot/Src/altimeter.cpp

^ the above is an example of the altimeter driver written using singletons. It shouldn't be too hard to pick up, and let any of us know if you have issues. Basically, calling one function should update the imu data struct and we can just use that one struct's data for our computation